### DISCUSSION TITLE
Getting started with software installation

### DISCUSSION THREAD

**Discussion by tnph** [OMOTE-Community/OMOTE-Firmware#34] [Q&A]
Yesterday I received 5 populated PCBs from PCBway, so now I want to install the software. I realise it's all still WIP but initially I just want to check the hardware works. 

I'm not a developer but my job is IT solution architect so technical and can code a bit.  I installed VS code and PlatformIO IDE but not sure of next steps.  I've been playing with Home assistant so know about flashing firmware etc...

**Comment by CoretechR** [comment]
Hi, you can just download the GitHub repository as a zip file and then open the Platformio folder in VS Code, which should recognize the project. If everything works, you should be able to press the program button at the bottom of the VS Code window to compile and upload the software.

**Reply by tnph** [reply]
Thanks @CoretechR that worked and I have the default pages displaying on the TFT, the battery works, it turns on when I pick it up etc - overall awesome!  

So next step is to configure it for my TV, Amp, Projector etc...  Where do I start?  Is the abstraction branch the place to start?

**Reply by CoretechR** [reply]
Great to hear that it works, @tnph !
Until the abstraction branch is complete and stable, I would stick with the base branch. You can add a new tab to the tabview and add your own devices or modify the existing ones.

**Comment by tnph** [comment]
I'm very keen to have a go at adding new tabs and devices.  You make it sound so easy :-) 
Could you point me to the specific files/code that I need to modify as a starting point please.

**Reply by CoretechR** [reply]
With the base branch, everything is inside the platformio/sec/main.cpp file. You can start off by searching for tab1-tab4 inside the code. There you can see where the tabs are declared and what objects exist within these tabs.

**Comment by tnph** [comment]
Can anyone supply the code changes required to get the battery level display working?
